### PR TITLE
Stop pipelines concurrently

### PR DESCRIPTION
**Pull request by danielmarbach**
Connects to #3447

Aligns with [Lifecycle of startable and stoppable components](https://github.com/Particular/PlatformDevelopment/pull/616)

@Particular/nservicebus-maintainers please review and merge

### CODE REVIEW

**Comment by timbussmann** [comment]
:+1:

**Comment by bording** [comment]
:+1: 

Using `Select()` like that to get a collection of tasks is not something I've seen before. I'm going to remember that one!

**Comment by bording** [comment]
I was going to go ahead and merge this, but I realized I wasn't sure what the merging policy was. I know we want no-ff merges, but should branches always be rebased to latest develop before merging?

**Comment by danielmarbach** [comment]
We usually use the mergebutton. If it is green, hit it ;) rebase is only
necessary when a rebase is needed (button grey). We are pretty pragmatic
about our history ;)

File: src/NServiceBus.Core/Pipeline/PipelineCollection.cs [hunk 1]
```diff
@@ -4,12 +4,10 @@ namespace NServiceBus
     using System.Collections.Generic;
     using System.Linq;
     using System.Threading.Tasks;
-    using Logging;
+    using NServiceBus.Logging;
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Pipeline/PipelineCollection.cs]: Side note: We need to tweak this setting in R#/VS to not add uneeded NS qualifications.  I'll raise a new issue for this

**Pull request merged**